presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -106,9 +106,13 @@ public NativeExecutionTask(
                 this.workerClient,
                 this.executor,
                 taskManagerConfig.getInfoUpdateInterval());
-        this.taskResultFetcher = new HttpNativeExecutionTaskResultFetcher(
-                updateScheduledExecutor,
-                this.workerClient);
+        if (!shuffleWriteInfo.isPresent()) {
+            this.taskResultFetcher = Optional.of(new HttpNativeExecutionTaskResultFetcher(
```

Comment:
Should we start the result fetcher only once the task has transitioned into `RUNNING` state. 
This is to avoid polling when the task has `FAILED`/`ABORTED` ?